presto-main/src/main/java/io/prestosql/metadata/MetadataManager.java

```diff
@@ -390,30 +398,39 @@ public boolean catalogExists(Session session, String catalogName)
             return Optional.empty();
         }
 
-        return Optional.of(new TableLayoutResult(fromConnectorLayout(connectorId, transaction, layouts.get(0).getTableLayout()), layouts.get(0).getUnenforcedConstraint()));
+        ConnectorTableLayout tableLayout = layouts.get(0).getTableLayout();
```

Comment:
Should we enforce connectors return only single layout?

Reply:
Good idea. I'll add that.